### PR TITLE
Develop1

### PR DESCRIPTION
**Pull request by BazilMuralidharan**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--{repo}--{owner}.hlx.live/
- After: https://<branch>--{repo}--{owner}.hlx.live/

### CODE REVIEW

**Review by SnehaBharathi22 — APPROVED** [review]
Approved

File: blocks/form/components/floatingFields.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    wrapper.dataset.empty = !inputField.value;
+  });
+  inputField?.addEventListener('blur', () => {
+    delete wrapper.dataset.active;
+    wrapper.dataset.empty = !inputField.value;
+  });
+  if (wrapper?.dataset) {
+    wrapper.dataset.empty = !inputField.value;
+  }
+  return panel;
```
Comment (vdua) [inline blocks/form/components/floatingFields.js:18]: for floating labels please follow the updated article
https://main--afb--adobe.hlx.live/docs/developer/floating-labels

File: common/executeinterfaceutils.js [hunk 2]
```diff
@@ -1,16 +1,311 @@
...
+  const date = new Date(inputDate);
+
+  const day = date.getDate();
+  const month = date.toLocaleString('default', { month: 'short' });
+  const year = date.getFullYear();
+
+  const formattedDate = `${day}-${month}-${year}`;
+
+  return formattedDate;
+};
```
Comment (vdua) [inline common/executeinterfaceutils.js:35]: support for formatting date is provided in the sdk. Any specific reason for implementing it here ?

File: common/executeinterfaceutils.js [hunk 3]
```diff
@@ -1,16 +1,311 @@
...
+      fullName,
+      panCheckFlag,
+      perAddressType: '2',
+      personalEmailId: personalDetails.personalEmailAddress.$value,
+      selfConfirmation: 'N',
+      addressEditFlag: isAddressEditFlag,
+      communicationAddress1: currentAddress.address1,
+      communicationAddress2: currentAddress.address2,
+      communicationCity: currentAddress.city,
+      dateOfBirth: formatDate(personalDetails.dobPersonalDetails.$value),
```
Comment (vdua) [inline common/executeinterfaceutils.js:124]: you can set a display format for the date field and use $displayValue.

File: common/executeinterfaceutils.js [hunk 4]
```diff
@@ -1,16 +1,311 @@
...
+ */
+const listNameOnCard = (globals) => {
+  const elementNameSelect = 'nameOnCardDropdown';
+  const { firstName, middleName, lastName } = currentFormContext.customerName;
+  const dropDownSelectField = globals.form.corporateCardWizardView.confirmCardPanel.cardBenefitsPanel.CorporatetImageAndNamePanel.nameOnCardDropdown;
+  const options = composeNameOption(firstName, middleName, lastName);
+  const initialValue = options[0]?.value;
+  setSelectOptions(options, elementNameSelect);
+  const setDropdownField = formUtil(globals, dropDownSelectField);
+  setDropdownField.setEnum(options, initialValue); // setting initial value
```
Comment (vdua) [inline common/executeinterfaceutils.js:202]: options should be an array of strings. Here it is an array of objects with value property

File: common/formutils.js [hunk 5]
```diff
@@ -154,6 +162,117 @@ const setDataAttributeOnClosestAncestor = (elementName, fieldValue, dataAttribut
...
+  currentMenuItem?.classList?.remove('wizard-menu-active-item');
+  navigateToMenuItem?.classList?.add('wizard-menu-active-item');
+  const event = new CustomEvent('wizard:navigate', {
+    detail: {
+      prevStep: { id: current?.id, index: parseInt(current?.dataset?.index || 0, 10) },
+      currStep: { id: navigateTo?.id, index: parseInt(navigateTo?.dataset?.index || 0, 10) },
+    },
+    bubbles: false,
+  });
+  navigateFrom?.dispatchEvent(event);
```
Comment (vdua) [inline common/formutils.js:255]: setFocus rule doesn't work ?

**Pull request closed without merging**